### ISSUE TITLE
Unexpected End of JSON Input

### ISSUE THREAD

**Issue by upstart-swiss** [SumoLogic/terraform-provider-sumologic#188]
during the refresh step, I'm consistently getting a
Error: unexpected end of JSON input

The TF_LOG=DEBUG is mostly useless, but here's some relevant stuff from TRACE
```
2021/04/30 08:53:19 [TRACE] eval: *terraform.EvalWriteState
2021/04/30 08:53:19 [TRACE] EvalWriteState: recording 1 dependencies for module.collectors.sumologic_http_source.searchresults20200916t122321_9920700_csv_1157794899
2021/04/30 08:53:19 [TRACE] EvalWriteState: writing current state object for module.collectors.sumologic_http_source.searchresults20200916t122321_9920700_csv_1157794899
2021/04/30 08:53:19 [TRACE] [walkRefresh] Exiting eval tree: module.collectors.sumologic_http_source.searchresults20200916t122321_9920700_csv_1157794899
2021/04/30 08:53:19 [TRACE] vertex "module.collectors.sumologic_http_source.searchresults20200916t122321_9920700_csv_1157794899": visit complete
2021/04/30 08:53:19 [TRACE] vertex "module.collectors.sumologic_http_source.searchresults20200916t122321_9920700_csv_1157794899": dynamic subgraph completed successfully
2021/04/30 08:53:19 [TRACE] vertex "module.collectors.sumologic_http_source.searchresults20200916t122321_9920700_csv_1157794899": visit complete
2021/04/30 08:53:19 [TRACE] vertex "module.collectors.sumologic_http_source.searchresults20200916t122321_9920700_csv_1157794899 (expand)": dynamic subgraph completed successfully
2021/04/30 08:53:19 [TRACE] vertex "module.collectors.sumologic_http_source.searchresults20200916t122321_9920700_csv_1157794899 (expand)": visit complete
2021/04/30 08:53:20 [TRACE] eval: *terraform.EvalWriteState
2021/04/30 08:53:20 [TRACE] EvalWriteState: recording 1 dependencies for module.collectors.sumologic_http_source.<REDACTED>_api_staging
2021/04/30 08:53:20 [TRACE] EvalWriteState: writing current state object for module.collectors.sumologic_http_source.<REDACTED>_api_staging
2021/04/30 08:53:20 [TRACE] [walkRefresh] Exiting eval tree: module.collectors.sumologic_http_source.<REDACTED>_api_staging
2021/04/30 08:53:20 [TRACE] vertex "module.collectors.sumologic_http_source.<REDACTED>_api_staging": visit complete
2021/04/30 08:53:20 [TRACE] vertex "module.collectors.sumologic_http_source.<REDACTED>_api_staging": dynamic subgraph completed successfully
2021/04/30 08:53:20 [TRACE] vertex "module.collectors.sumologic_http_source.<REDACTED>_api_staging": visit complete
2021/04/30 08:53:20 [TRACE] vertex "module.collectors.sumologic_http_source.<REDACTED>_api_staging (expand)": dynamic subgraph completed successfully
2021/04/30 08:53:20 [TRACE] vertex "module.collectors.sumologic_http_source.<REDACTED>_api_staging (expand)": visit complete
2021/04/30 08:53:20 [TRACE] eval: *terraform.EvalWriteState
2021/04/30 08:53:20 [TRACE] EvalWriteState: recording 1 dependencies for module.collectors.sumologic_http_source.<REDACTED>_api_prod
2021/04/30 08:53:20 [TRACE] EvalWriteState: writing current state object for module.collectors.sumologic_http_source.<REDACTED>_api_prod
2021/04/30 08:53:20 [TRACE] [walkRefresh] Exiting eval tree: module.collectors.sumologic_http_source.<REDACTED>_api_prod
2021/04/30 08:53:20 [TRACE] vertex "module.collectors.sumologic_http_source.<REDACTED>_api_prod": visit complete
2021/04/30 08:53:20 [TRACE] vertex "module.collectors.sumologic_http_source.<REDACTED>_api_prod": dynamic subgraph completed successfully
2021/04/30 08:53:20 [TRACE] vertex "module.collectors.sumologic_http_source.<REDACTED>_api_prod": visit complete
2021/04/30 08:53:20 [TRACE] vertex "module.collectors.sumologic_http_source.<REDACTED>_api_prod (expand)": dynamic subgraph completed successfully
2021/04/30 08:53:20 [TRACE] vertex "module.collectors.sumologic_http_source.<REDACTED>_api_prod (expand)": visit complete
2021/04/30 08:53:20 [TRACE] eval: *terraform.EvalWriteState
2021/04/30 08:53:20 [TRACE] EvalWriteState: recording 1 dependencies for module.collectors.sumologic_http_source.<REDACTED>_api_sandbox_102624242
2021/04/30 08:53:20 [TRACE] EvalWriteState: writing current state object for module.collectors.sumologic_http_source.<REDACTED>_api_sandbox_102624242
2021/04/30 08:53:20 [TRACE] [walkRefresh] Exiting eval tree: module.collectors.sumologic_http_source.<REDACTED>_api_sandbox_102624242
2021/04/30 08:53:20 [TRACE] vertex "module.collectors.sumologic_http_source.<REDACTED>_api_sandbox_102624242": visit complete
2021/04/30 08:53:20 [TRACE] vertex "module.collectors.sumologic_http_source.<REDACTED>_api_sandbox_102624242": dynamic subgraph completed successfully
2021/04/30 08:53:20 [TRACE] vertex "module.collectors.sumologic_http_source.<REDACTED>_api_sandbox_102624242": visit complete
2021/04/30 08:53:20 [TRACE] vertex "module.collectors.sumologic_http_source.<REDACTED>_api_sandbox_102624242 (expand)": dynamic subgraph completed successfully
2021/04/30 08:53:20 [TRACE] vertex "module.collectors.sumologic_http_source.<REDACTED>_api_sandbox_102624242 (expand)": visit complete
2021/04/30 08:53:20 [TRACE] eval: *terraform.EvalWriteState
2021/04/30 08:53:20 [TRACE] EvalWriteState: recording 1 dependencies for module.collectors.sumologic_http_source.spark_staging_logs_957270217
2021/04/30 08:53:20 [TRACE] EvalWriteState: writing current state object for module.collectors.sumologic_http_source.spark_staging_logs_957270217
2021/04/30 08:53:20 [TRACE] [walkRefresh] Exiting eval tree: module.collectors.sumologic_http_source.spark_staging_logs_957270217
2021/04/30 08:53:20 [TRACE] vertex "module.collectors.sumologic_http_source.spark_staging_logs_957270217": visit complete
2021/04/30 08:53:20 [TRACE] vertex "module.collectors.sumologic_http_source.spark_staging_logs_957270217": dynamic subgraph completed successfully
2021/04/30 08:53:20 [TRACE] vertex "module.collectors.sumologic_http_source.spark_staging_logs_957270217": visit complete
2021/04/30 08:53:20 [TRACE] vertex "module.collectors.sumologic_http_source.spark_staging_logs_957270217 (expand)": dynamic subgraph completed successfully
2021/04/30 08:53:20 [TRACE] vertex "module.collectors.sumologic_http_source.spark_staging_logs_957270217 (expand)": visit complete
2021/04/30 08:53:21 [TRACE] eval: *terraform.EvalWriteState
2021/04/30 08:53:21 [TRACE] EvalWriteState: recording 1 dependencies for module.collectors.sumologic_http_source.<REDACTED>_web_production_386100814
2021/04/30 08:53:21 [TRACE] EvalWriteState: writing current state object for module.collectors.sumologic_http_source.<REDACTED>_web_production_386100814
2021/04/30 08:53:21 [TRACE] [walkRefresh] Exiting eval tree: module.collectors.sumologic_http_source.<REDACTED>_web_production_386100814
2021/04/30 08:53:21 [TRACE] vertex "module.collectors.sumologic_http_source.<REDACTED>_web_production_386100814": visit complete
2021/04/30 08:53:21 [TRACE] vertex "module.collectors.sumologic_http_source.<REDACTED>_web_production_386100814": dynamic subgraph completed successfully
2021/04/30 08:53:21 [TRACE] vertex "module.collectors.sumologic_http_source.<REDACTED>_web_production_386100814": visit complete
2021/04/30 08:53:21 [TRACE] vertex "module.collectors.sumologic_http_source.<REDACTED>_web_production_386100814 (expand)": dynamic subgraph completed successfully
2021/04/30 08:53:21 [TRACE] vertex "module.collectors.sumologic_http_source.<REDACTED>_web_production_386100814 (expand)": visit complete
2021/04/30 08:53:21 [TRACE] eval: *terraform.EvalWriteState
2021/04/30 08:53:21 [TRACE] EvalWriteState: recording 1 dependencies for module.collectors.sumologic_http_source.<REDACTED>_web_production_console_collector_228991665
2021/04/30 08:53:21 [TRACE] EvalWriteState: writing current state object for module.collectors.sumologic_http_source.<REDACTED>_web_production_console_collector_228991665
2021/04/30 08:53:21 [TRACE] [walkRefresh] Exiting eval tree: module.collectors.sumologic_http_source.<REDACTED>_web_production_console_collector_228991665
2021/04/30 08:53:21 [TRACE] vertex "module.collectors.sumologic_http_source.<REDACTED>_web_production_console_collector_228991665": visit complete
2021/04/30 08:53:21 [TRACE] vertex "module.collectors.sumologic_http_source.<REDACTED>_web_production_console_collector_228991665": dynamic subgraph completed successfully
2021/04/30 08:53:21 [TRACE] vertex "module.collectors.sumologic_http_source.<REDACTED>_web_production_console_collector_228991665": visit complete
2021/04/30 08:53:21 [TRACE] vertex "module.collectors.sumologic_http_source.<REDACTED>_web_production_console_collector_228991665 (expand)": dynamic subgraph completed successfully
2021/04/30 08:53:21 [TRACE] vertex "module.collectors.sumologic_http_source.<REDACTED>_web_production_console_collector_228991665 (expand)": visit complete
2021/04/30 08:53:21 [TRACE] eval: *terraform.EvalWriteState
2021/04/30 08:53:21 [TRACE] EvalWriteState: recording 1 dependencies for module.collectors.sumologic_http_source.<REDACTED>_api_staging_2_101996514
2021/04/30 08:53:21 [TRACE] EvalWriteState: writing current state object for module.collectors.sumologic_http_source.<REDACTED>_api_staging_2_101996514
2021/04/30 08:53:21 [TRACE] [walkRefresh] Exiting eval tree: module.collectors.sumologic_http_source.<REDACTED>_api_staging_2_101996514
2021/04/30 08:53:21 [TRACE] vertex "module.collectors.sumologic_http_source.<REDACTED>_api_staging_2_101996514": visit complete
2021/04/30 08:53:21 [TRACE] vertex "module.collectors.sumologic_http_source.<REDACTED>_api_staging_2_101996514": dynamic subgraph completed successfully
2021/04/30 08:53:21 [TRACE] vertex "module.collectors.sumologic_http_source.<REDACTED>_api_staging_2_101996514": visit complete
2021/04/30 08:53:21 [TRACE] vertex "module.collectors.sumologic_http_source.<REDACTED>_api_staging_2_101996514 (expand)": dynamic subgraph completed successfully
2021/04/30 08:53:21 [TRACE] vertex "module.collectors.sumologic_http_source.<REDACTED>_api_staging_2_101996514 (expand)": visit complete
2021/04/30 08:53:21 [TRACE] eval: *terraform.EvalWriteState
2021/04/30 08:53:21 [TRACE] EvalWriteState: recording 1 dependencies for module.collectors.sumologic_http_source.<REDACTED>_poc
2021/04/30 08:53:21 [TRACE] EvalWriteState: writing current state object for module.collectors.sumologic_http_source.<REDACTED>_poc
2021/04/30 08:53:21 [TRACE] [walkRefresh] Exiting eval tree: module.collectors.sumologic_http_source.<REDACTED>_poc
2021/04/30 08:53:21 [TRACE] vertex "module.collectors.sumologic_http_source.<REDACTED>_poc": visit complete
2021/04/30 08:53:21 [TRACE] vertex "module.collectors.sumologic_http_source.<REDACTED>_poc": dynamic subgraph completed successfully
2021/04/30 08:53:21 [TRACE] vertex "module.collectors.sumologic_http_source.<REDACTED>_poc": visit complete
2021/04/30 08:53:21 [TRACE] vertex "module.collectors.sumologic_http_source.<REDACTED>_poc (expand)": dynamic subgraph completed successfully
2021/04/30 08:53:21 [TRACE] vertex "module.collectors.sumologic_http_source.<REDACTED>_poc (expand)": visit complete
2021/04/30 08:53:22 [TRACE] eval: *terraform.EvalWriteState
2021/04/30 08:53:22 [TRACE] EvalWriteState: recording 1 dependencies for module.collectors.sumologic_http_source.<REDACTED>_staging_auto_1138925242
2021/04/30 08:53:22 [TRACE] EvalWriteState: writing current state object for module.collectors.sumologic_http_source.<REDACTED>_staging_auto_1138925242
2021/04/30 08:53:22 [TRACE] [walkRefresh] Exiting eval tree: module.collectors.sumologic_http_source.<REDACTED>_staging_auto_1138925242
2021/04/30 08:53:22 [TRACE] vertex "module.collectors.sumologic_http_source.<REDACTED>_staging_auto_1138925242": visit complete
2021/04/30 08:53:22 [TRACE] vertex "module.collectors.sumologic_http_source.<REDACTED>_staging_auto_1138925242": dynamic subgraph completed successfully
2021/04/30 08:53:22 [TRACE] vertex "module.collectors.sumologic_http_source.<REDACTED>_staging_auto_1138925242": visit complete
2021/04/30 08:53:22 [TRACE] vertex "module.collectors.sumologic_http_source.<REDACTED>_staging_auto_1138925242 (expand)": dynamic subgraph completed successfully
2021/04/30 08:53:22 [TRACE] vertex "module.collectors.sumologic_http_source.<REDACTED>_staging_auto_1138925242 (expand)": visit complete
2021/04/30 08:53:22 [TRACE] dag/walk: visiting "module.collectors (close)"
2021/04/30 08:53:22 [TRACE] dag/walk: visiting "module.collectors.provider[\"registry.terraform.io/sumologic/sumologic\"] (close)"
2021/04/30 08:53:22 [TRACE] vertex "module.collectors (close)": starting visit (*terraform.nodeCloseModule)
2021/04/30 08:53:22 [TRACE] vertex "module.collectors (close)": evaluating
2021/04/30 08:53:22 [TRACE] [walkRefresh] Entering eval tree: module.collectors (close)
2021/04/30 08:53:22 [TRACE] eval: *terraform.EvalSequence
2021/04/30 08:53:22 [TRACE] eval: terraform.EvalNoop
2021/04/30 08:53:22 [TRACE] [walkRefresh] Exiting eval tree: module.collectors (close)
2021/04/30 08:53:22 [TRACE] vertex "module.collectors (close)": visit complete
2021/04/30 08:53:22 [TRACE] vertex "module.collectors.provider[\"registry.terraform.io/sumologic/sumologic\"] (close)": starting visit (*terraform.graphNodeCloseProvider)
2021/04/30 08:53:22 [TRACE] vertex "module.collectors.provider[\"registry.terraform.io/sumologic/sumologic\"] (close)": evaluating
2021/04/30 08:53:22 [TRACE] [walkRefresh] Entering eval tree: module.collectors.provider["registry.terraform.io/sumologic/sumologic"] (close)
2021/04/30 08:53:22 [TRACE] eval: *terraform.EvalCloseProvider
2021/04/30 08:53:22 [TRACE] GRPCProvider: Close
2021-04-30T08:53:22.241-0700 [DEBUG] plugin: plugin process exited: path=.terraform/plugins/registry.terraform.io/sumologic/sumologic/2.9.1/darwin_amd64/terraform-provider-sumologic_v2.9.1 pid=7262
2021-04-30T08:53:22.241-0700 [DEBUG] plugin: plugin exited
2021/04/30 08:53:22 [TRACE] [walkRefresh] Exiting eval tree: module.collectors.provider["registry.terraform.io/sumologic/sumologic"] (close)
2021/04/30 08:53:22 [TRACE] vertex "module.collectors.provider[\"registry.terraform.io/sumologic/sumologic\"] (close)": visit complete
2021/04/30 08:53:22 [TRACE] dag/walk: upstream of "root" errored, so skipping

Error: unexpected end of JSON input


2021-04-30T08:53:22.244-0700 [DEBUG] plugin: plugin process exited: path=.terraform/plugins/registry.terraform.io/sumologic/sumologic/2.9.1/darwin_amd64/terraform-provider-sumologic_v2.9.1 pid=7261

```

**Comment by vsinghal13** [comment]
@upstart-swiss Could you please give a brief background on what you are trying to do with the http_resource and do you see any error when you apply the config? also, could you attach the terraform config file?

**Comment by upstart-swiss** [comment]
You mean http_source?

I can't share a ton here because of company confidentiality. we do have a joint slack channel between Sumo/my group (message freno) and I can share more details on there if you need more details

**Comment by upstart-swiss** [comment]
Did digging and discovery, a lot more details:

If a folder is created, and another account (we separate our RO and RW accounts for stages in the CI pipeline) cannot see that content, the terraform plan (refreshing that content) fails.

Desired behavior should be a 403 + a meaningful error that says the user running the plan cannot view the resource.

**Comment by devanbenz** [comment]
@upstart-swiss did you end up resolving this error? I just had something very similar happen to me as well. We are trying to use the sumo logic provider and during folder creation, I get an error like so: 

` Error: unexpected end of JSON input
│
│   with module.sumologic.sumologic_folder.per_environment,
│   on modules/sumologic/folders.tf line 4, in resource "sumologic_folder" "per_environment":
│    4: resource "sumologic_folder" "per_environment" {`

**Comment by emichaeli** [comment]
@upstart-swiss thanks for reporting this and giving clear explanation of what happened. I created the above PR to address this issue.

> Desired behavior should be a 403 + a meaningful error that says the user running the plan cannot view the resource.

We'll actually respond with a `Content with the given ID does not exist` error. In this case, if the API were to return a `403` when users try to read a specific resource they don't have permissions to see, then we'll reveal the fact that this resource exists. I.e. if the resource exists they'll get a `403`, and if it doesn't, a `404`. So in the case of a read where users have access to some resources but not all, we'll return `200` for all resources they're allowed to read and `404` for all resources that either do not exist or they have no permissions to read.

@devanbenz let me followup in https://github.com/SumoLogic/terraform-provider-sumologic/issues/317 about your specific issue.

**Comment by emichaeli** [comment]
Fixed in with provider `v2.11.4` with https://github.com/SumoLogic/terraform-provider-sumologic/pull/319
